### PR TITLE
Updates to allow using task_storage.RedisS3 for payloads

### PR DESCRIPTION
**Pull request by yadudoc**
Support for storing larger payloads in S3 via RedisS3Storage.

Please note that this PR currently runs the risk of receiving tasks launched by an older version of funcx-web-service which does not follow the `payload_reference` convention expected here and cause some breakage. 

[sc-11106]

### CODE REVIEW

**Comment by yadudoc** [comment]
@sirosen is working on moving `RedisTask` over to `funcx-common` with additional logic for the case described above where a task being processed by the `funcx-forwarder` was created by an older `funcx-web-service` and is missing the `payload_reference` field. I'm planning to hold this PR and the matching one on the funcx-web-service (https://github.com/funcx-faas/funcx-web-service/pull/282) until those updates are available in `funcx-common`.

File: funcx_forwarder/tasks.py [hunk 1]
```diff
@@ -45,7 +45,10 @@ class RedisTask(TaskProtocol, metaclass=HasRedisFieldsMeta):
     function_id = RedisField()
     endpoint = t.cast(str, RedisField())
     container = RedisField()
-    payload = RedisField(serde=JSON_SERDE)
+    payload = t.cast(t.Optional[str], RedisField(serde=JSON_SERDE))
```
Comment (joshbryan-globus) [inline funcx_forwarder/tasks.py:48]: If we release like this and eventually change this to not use the JSON_SERDE, we won't be able to use the existance of payload_reference to determine whether or not to deserialize.  We could do it with an additional field in the payload reference, but just the existance won't be enough.  I'm fine with that, but just want to call it out before we go forward with this.

Comment (yadudoc) [reply funcx_forwarder/tasks.py:48]: I believe the plan is to use a new field in the reference. -> https://github.com/funcx-faas/funcx-common/pull/15#discussion_r777721291

**Review by joshbryan-globus — APPROVED** [review]

File: funcx_forwarder/tasks.py (hunk 1 above)
Comment (sirosen) [reply funcx_forwarder/tasks.py:48]: I missed this comment, but where Yadu pointed shows my thoughts on this.
By not solving it today, we keep the work simpler. The hard part will be making the `serde` object refer back to the owner of the `RedisField` anyway. Once we do that, doing it by existence of the field or a value inside of it is trivial. I'll mark this resolved.

**Review by sirosen — APPROVED** [review]

**Pull request merged**